minimal-notebook/Dockerfile

```diff
@@ -43,6 +47,16 @@ RUN conda install --yes \
     terminado \
     && conda clean -yt
 
+# Create jovyan user with UID=1000 and in the 'users' group
+# Grant ownership over the conda dir and home dir, but stick the group as root.
+RUN useradd -m -s /bin/bash -N -u $NB_UID $NB_USER && \
+    chmod g+s /home/$NB_USER && \
+    mkdir /home/$NB_USER/work && \
+    mkdir /home/$NB_USER/.jupyter && \
+    mkdir /home/$NB_USER/.local && \
+    chown -R $NB_USER:root $CONDA_DIR && \
+    chown -R $NB_USER:root /home/$NB_USER
```

Comment:
Exactly what I wanted to ask you. :) The `-N` on useradd puts the `jovyan` user in the default `users` group (GID=100). Anything that user touches in a host mount home or work dir will be accessible by GID=100 on the host unless the root of the mount has proper permissions. To head off any "Oops, I didn't think to protect against a container / host group overlap as well as UID overlap", I figured root was a safe bet.

But I agree it's quirky. I'm open to suggestions here.